scripts/version_maker.sh

```diff
@@ -18,63 +22,126 @@ function yaml() {
 
-        echo $newversion
-        # if  version is interger or float (semversion)
-        sudo python scripts/validate_version.py $newversion
+        temp=$(echo ${file::-18}) # This statement will scrap last 18 character which will be .basetemplate.yaml
+        if [[ $kind == "ChartServiceVersion" ]]; then
+            oldversionfile=$temp'.yaml'
+            echo $oldversionfile
+            eval $(yaml_parser $oldversionfile)
 
-        if [[ $? == 0 ]]; then
-                temp=$(echo ${file::-18})
-            if [[ $kind == "ChartServiceVersion" ]]; then
-                # echo $temp
-                oldversionfile=$temp'.yaml'
-                echo $oldversionfile
-                eval $(yaml $oldversionfile)
-
-                echo $?
-                if [[ $? == 0 ]]; then
-                    oldversion=$spec_version
-                    echo $oldversion
+            if [[ $? == 0 ]]; then
+                oldversion=$spec_version
+                # This python script checks and validate the version
+                sudo python3 scripts/validate_version.py $oldversion $newversion
 
-                    `sed -i  "s/$oldversion/$newversion/" $oldversionfile` &&
-                    `sed -i  "s/version:[[:space:]]*$newversion/version: {{ VERSION }}/" $file`
+                if [[ $? == 0 ]]; then
+                    `sed -i "s/$oldversion/$newversion/" $oldversionfile` &&
+                    `sed -i "s/version:[[:space:]]*$newversion/version: {{ VERSION }}/" $file`
+                    echo "$file's version updated from $oldversion to $newversion"
                 fi
+            fi
+        elif [ $kind == "ChaosExperiment" ]; then
 
-            elif [ $kind == "ChaosExperiment" ]; then
-                # echo $temp
-                oldversionfile=$temp'.version.yaml'
-                echo $oldversionfile
-                eval $(yaml $oldversionfile)
+            oldversionfile=$temp'.yaml'
+            echo $oldversionfile
+            eval $(yaml_parser $oldversionfile)
 
-                if [[ $? == 0 ]]; then
-                    oldversion=$metadata_version
-                    echo $oldversion
+            if [[ $? == 0 ]]; then
+                oldversion=$metadata_version
+
+                # This python script checks and validate the version
+                sudo python3 scripts/validate_version.py $oldversion $newversion
 
-                    `sed -i  "s/$oldversion/$newversion/" $oldversionfile`
+                if [[ $? == 0 ]]; then
+                    `sed -i "s/$oldversion/$newversion/" $oldversionfile` &&
                     `sed -i "s/version:[[:space:]]*$newversion/version: {{ VERSION }}/" $file`
+                    echo "$file's version updated from $oldversion to $newversion"
                 fi
             fi
         fi
     fi
+}
+
+# compare and retrive the changed files
+check_diff=`git diff ${second_last_commit_hash} --name-only`
+files=$(echo $check_diff | tr " " "\n")
```

Comment:
possible comment: convert a space-separated list of filenames changed in previous commit to an array.